src/main/java/org/opensearch/neuralsearch/query/NeuralSparseTwoPhaseParameters.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.opensearch.Version;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.ParsingException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.neuralsearch.settings.NeuralSearchSettings;
+import org.opensearch.neuralsearch.util.NeuralSearchClusterUtil;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Represents the parameters for neural_sparse two-phase process.
+ * This class encapsulates settings related to window size expansion, pruning ratio, and whether the two-phase search is enabled.
+ * It includes mechanisms to update settings from the cluster dynamically.
+ */
+@Getter
+@Setter
+@Accessors(chain = true, fluent = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class NeuralSparseTwoPhaseParameters implements Writeable {
+    public static volatile Float DEFAULT_WINDOW_SIZE_EXPANSION;
+    public static volatile Float DEFAULT_PRUNING_RATIO;
+    public static volatile Boolean DEFAULT_ENABLED;
+    public static volatile Integer MAX_WINDOW_SIZE;
```

Comment:
why public and volatile?

Reply:
and if field is static and final then the name should be all caps. But that is not the case here.

Reply:
> why public and volatile?

Using a volatile field and settings update consumer is a common approach in OpenSearch to get real-time cluster settings from memory, without accessing the cluster settings object. You can just find many references in other plugins.

The public can be removed.